ntcore/src/main/native/include/networktables/StructArrayTopic.h

```diff
@@ -0,0 +1,576 @@
+   * timestamp of 0.
+   *
+   * @return timestamped value
+   */
+  TimestampedValueType GetAtomic() const { return GetAtomic(m_defaultValue); }
+
+  /**
+   * Get the last published value along with its timestamp.
+   * If no value has been published, returns the passed defaultValue and a
+   * timestamp of 0.
+   *
+   * @param defaultValue default value to return if no value has been published
+   * @return timestamped value
+   */
+  template <typename U>
+#if __cpp_lib_ranges >= 201911L
+  requires std::ranges::range<U> &&
+      std::convertible_to<std::ranges::range_value_t<U>, T>
+#endif
+          TimestampedValueType GetAtomic(U&& defaultValue)
+  const {
+    wpi::SmallVector<uint8_t, 128> buf;
+    TimestampedRawView view = ::nt::GetAtomicRaw(m_subHandle, buf, {});
+    if (view.value.size() == 0 || (view.value.size() % S::kSize) != 0) {
+      return {0, 0, std::forward<U>(defaultValue)};
+    }
+    TimestampedValueType rv{view.time, view.serverTime, {}};
+    rv.value.reserve(view.value.size() / S::kSize);
+    for (auto in = view.value.begin(), end = view.value.end(); in != end;
+         in += S::kSize) {
+      rv.value.emplace_back(
+          S::Unpack(std::span<const uint8_t, S::kSize>{in, in + S::kSize}));
+    }
+    return rv;
+  }
+
+  /**
+   * Get the last published value along with its timestamp.
+   * If no value has been published, returns the passed defaultValue and a
+   * timestamp of 0.
+   *
+   * @param defaultValue default value to return if no value has been published
+   * @return timestamped value
+   */
+  TimestampedValueType GetAtomic(std::span<const T> defaultValue) const {
+    wpi::SmallVector<uint8_t, 128> buf;
+    TimestampedRawView view = ::nt::GetAtomicRaw(m_subHandle, buf, {});
+    if (view.value.size() == 0 || (view.value.size() % S::kSize) != 0) {
+      return {0, 0, {defaultValue.begin(), defaultValue.end()}};
+    }
+    TimestampedValueType rv{view.time, view.serverTime, {}};
+    rv.value.reserve(view.value.size() / S::kSize);
+    for (auto in = view.value.begin(), end = view.value.end(); in != end;
+         in += S::kSize) {
+      rv.value.emplace_back(
+          S::Unpack(std::span<const uint8_t, S::kSize>{in, in + S::kSize}));
+    }
+    return rv;
+  }
+
+  /**
+   * Get an array of all value changes since the last call to ReadQueue.
+   * Also provides a timestamp for each value.
+   *
+   * @note The "poll storage" subscribe option can be used to set the queue
+   *     depth.
+   *
+   * @return Array of timestamped values; empty array if no new changes have
+   *     been published since the previous call.
+   */
+  std::vector<TimestampedValueType> ReadQueue() {
+    auto raw = ::nt::ReadQueueRaw(m_subHandle);
+    std::vector<TimestampedValueType> rv;
+    rv.reserve(raw.size());
+    for (auto&& r : raw) {
+      if (r.value.size() == 0 || (r.value.size() % S::kSize) != 0) {
+        continue;
+      }
```

Comment:
Is this desired behavior (completely and silently skipping over invalid values)? Should this be documented?